applications/minotari_console_wallet/src/grpc/wallet_grpc_server.rs

```diff
@@ -202,6 +213,110 @@ impl WalletGrpcServer {
             .unwrap_or_default();
         Ok(self.rules.consensus_constants(height))
     }
+
+    /// Calculate PayRefs for a transaction's outputs using existing wallet output data
+    async fn calculate_payment_references_for_transaction(
+        &self,
+        output_commitments: &[Vec<u8>],
+        mined_height: u64,
+    ) -> Vec<Vec<u8>> {
+        calculate_payment_references_impl(
+            self.get_output_manager_service(),
+            output_commitments,
+            mined_height,
```

Comment:
```suggestion
    async fn calculate_payment_references_for_transaction(
        output_manager: &OutputManagerHandle,
        output_commitments: &[Vec<u8>],
        mined_height: u64,
    ) -> Vec<Vec<u8>> {
        calculate_payment_references_impl(
            output_manager,
            output_commitments,
            mined_height,
```
By doing this, you don't have to pass around the grpc wallet service